### PR TITLE
fix issue #340

### PR DESCRIPTION
**Pull request by felixmusil**
Change how the inputs are checked in `convert_to_structure_list` to keep the compatibility with various versions of ASE.

### CODE REVIEW

**Comment by max-veit** [comment]
This seems OK to me, but why use assertions to check for the type of atoms list? I would rather have it check first for ASE Atoms, then for Iterable, and if neither of the two applies, then fail with an informative error message (saying e.g. that you need to pass either Atoms or a list of Atoms).

**Comment by felixmusil** [comment]
It's true that it is typically done as you outline in the codebase but I feel this makes the docstring quite redundant in this case and thus heavier to extend/maintain later on. The `assert` seems clear enough to me and if not then the docstring is there to help. What do you think?

**Comment by max-veit** [comment]
Assertion errors are not clear as error messages; they're typically used instead _by programmers_ to debug conditions _they_ think should hold true, no matter what the user input. A clear error message, on the other hand, does not make the docstring redundant, instead telling the user what exactly they did wrong. This will make maintenance _much easier_ later on because if one of us passes the wrong thing by mistake, we know exactly what's going on and don't have to waste 15 minutes figuring out where in the code their error got spit out from.

File: bindings/rascal/neighbourlist/structure_manager.py [hunk 1]
```diff
@@ -228,8 +228,23 @@ def get_neighbourlist(structure, options):
...
+    Returns
+    -------
+    structure_list : AtomicStructureList
+
+    """
+    if isinstance(frames, Atoms):
         frames = [frames]
+
+    assert isinstance(frames, list)
+    assert isinstance(frames[0], Atoms) or isinstance(frames[0], dict)
```
Comment (max-veit) [inline bindings/rascal/neighbourlist/structure_manager.py:246]:
```suggestion
    elif not isinstance(frames, Iterable):
        raise ValueError("Must pass either an ase.Atoms object or an iterable")
```
The check of whether the objects being iterated over are Atoms (or dict, or similar) is handled later on (see line 257 in this version).

File: bindings/rascal/neighbourlist/structure_manager.py [hunk 2]
```diff
@@ -1,10 +1,10 @@
-from collections.abc import Iterable
 from functools import reduce
```
Comment (max-veit) [inline bindings/rascal/neighbourlist/structure_manager.py:1]:
```suggestion
from collections.abc import Iterable
from functools import reduce
```

**Review by max-veit — CHANGES_REQUESTED** [review]
Here, it's very simple but makes for a much better user (and developer) experience.

**Comment by max-veit** [comment]
> Assertions are a systematic way to check that the **internal state** of a program is as the programmer expected, with the goal of catching bugs. In particular, they're good for **catching false assumptions** that were made while writing the code, or abuse 

https://wiki.python.org/moin/UsingAssertionsEffectively (emphasis mine)

**Review by max-veit — APPROVED** [review]
Ok, all good from my end, provided the tests pass **and** that you squash these into one commit with an informative message.

**Pull request merged**